pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1228,4 +1228,10 @@ self: super: {
     '';
   });
 
+  # https://github.com/brendanhay/amazonka/commit/657b70d174fe5cb61e56cb8b9c5e57f1ec216f2b
+  amazonka = appendPatch super.amazonka ./patches/amazonka-Allow-http-client-0.6.patch;
```

Comment:
@jbgi Generally, people don't like to carry around patches in nixpkgs, but instead pull them from upstream.  

Could you change this to pull the patch from upstream, similar to below:

https://github.com/NixOS/nixpkgs/pull/68935#discussion_r324888857

Reply:
I see. Though in this case there is a single [commit patch](https://github.com/brendanhay/amazonka/commit/657b70d174fe5cb61e56cb8b9c5e57f1ec216f2b.patch) that need to be split to apply to each package. Is there a precedent in nixpkgs of a similar situation?

Reply:
@jbgi Ah, I see.  I don't know what to do in that situation.  I'd say it looks fine as-is.